src/test/java/org/jenkinsci/plugins/workflow/support/steps/ExecutorStepTest.java

```diff
@@ -890,6 +894,34 @@ String n(Run<?, ?> b, String label) {
             b3.doStop();
         });
     }
+
+    /**
+     * @see PipelineOnlyTaskDispatcher
+     */
+    @Issue("JENKINS-53837")
+    @Test public void queueTaskOwnerCorrectWhenRestarting() {
+        story.then(r -> {
+            WorkflowJob p = r.createProject(WorkflowJob.class, "p1");
+            p.setDefinition(new CpsFlowDefinition("node {\n" +
+                    "  semaphore('wait')\n" +
+                    "}", true));
+            WorkflowRun b = p.scheduleBuild2(0).waitForStart();
+            SemaphoreStep.waitForStart("wait/1", b);
+        });
+        story.then(r -> {
+            WorkflowJob p = r.jenkins.getItemByFullName("p1", WorkflowJob.class);
+            WorkflowRun b = p.getBuildByNumber(1);
+            SemaphoreStep.waitForStart("wait/1", b);
+            SemaphoreStep.success("wait/1", null);
+            while (b.isBuilding()) {
```

Comment:
A bit convoluted. I would think it would suffice to say something like

```java
SemaphoreStep.success("wait/1", null);
r.waitForCompletion(b);
r.assertBuildStatusSuccess(b);
r.assertLogNotContains("Non-Pipeline tasks are forbidden!", b);
```

Reply:
Yeah, this was helpful for testing so that the test would fail immediately rather than timing out waiting, but I will clean it up now.